src/util/ieee_float.cpp

```diff
@@ -88,25 +87,33 @@ u32 f32Tou32Slow(f32 f)
 	return signbit | (exp << 23) | (imant & 0x7FFFFFUL);
 }
 
-
-// Test the functions.
 // This test needs the following requisites in order to work:
-// - The compiler must support u32.
 // - The float type must be a 32 bits IEEE-754 single-precision float.
 // - The endianness of f32s and integers must match.
 FloatType getFloatSerializationType()
 {
+	const f32 cf = -22220490.f;
+	const u32 cu = 0xCBA98765UL;
+	if (std::numeric_limits<f32>::is_iec559 && sizeof(cf) == 4 &&
+			sizeof(cu) == 4 && !memcmp(&cf, &cu, 4)) {
+		// u32Tof32Slow and f32Tou32Slow are not needed, use memcpy
+		return FLOATTYPE_SYSTEM;
+	}
+
+	// Run quick tests to ensure the custom functions provide acceptable results
+	warningstream << "floatSerialization: f32 and u32 endianness are "
+		"not equal or machine is not IEEE-754 compliant" << std::endl;
```

Comment:
I still think that everything beyond this line (except the last line) belongs in the unit test rather than here. Also, maybe add "Using conversion routines - performance will be affected" in the message?